File: Revisao/incremental7/CartaoBasico.java
class CartaoBasico extends CartaoDeCredito {

    public CartaoBasico(String nomeTitular, String numero, float limite, Fatura fatura) {
        super(nomeTitular, numero, limite, fatura);
    }

    @Override
    public void realizarCompra(float valor) {
        if (!ativo) {
            System.out.println("Cartão inativo. Transação não permitida.");
            return;
        }
        if (saldo + valor <= limite) {
            saldo += valor;
            Transacao transacao = new Transacao(new Date(), valor, "Compra básica", "compra");
            adicionarTransacao(transacao);
            System.out.println("Compra de R$" + valor + " realizada com sucesso no Cartão Básico.");
        } else {
            System.out.println("Compra não autorizada. Limite insuficiente.");
        }
    }
}

File: Revisao/python/cartaoBasico.py

import datetime
from Revisao.python import transacao
from Revisao.python.transacao import Transacao


class CartaoBasico():
    def realizar_compra(self, valor):
        if not self.ativo:
            print("Cartão desativado")
            return
        if self.saldo + valor <= self.limite:
            self.saldo += valor
            trans = Transacao(datetime.now(), valor, "compra", "credito")
            self.adicionar_transacao(transacao)
            print(f"Commpra de R${valor} realizada com sucesso")
        else:
            print("Saldo insuficiente")
            